Scheduled DB Maintainance (#195)
FlowFuse/website: src/blog/2022/10/db-migraton-01.md

> # Planned Database Migration

Reviewer: Needs the usual blog-post header meta-data adding

> As part of an on going security [review](https://flowforge.com/product/security/#data-at-rest) of the FlowForge Managed offering we discovered that the backend database was not using encrypted storage. In keeping with industry best practices we we plan to migrate the database to a new instance using encrypted at rest storage.

Reviewer: ```suggestion
As part of an on going security [review](https://flowforge.com/product/security/#data-at-rest) of the FlowForge Managed offering we discovered that the backend database was not using encrypted storage. In keeping with industry best practices we plan to migrate the database to a new instance using encrypted at rest storage.
```

## Impact

To do this cleanly we are going to need to schdule a short period of downtime for the core FlowForge application. During this downtime the following functions will not be available

- Creating/deleting FlowForge Projects
- Creating/deleting FlowForge Users
- Deploying changes to flows running in Projects

> Existing FlowForge Projects will continue to run as notmal during the downtime.

Reviewer: ```suggestion
Existing FlowForge Projects will continue to run as normal during the downtime.
```

---
title: "Scheduled maintenance: Database encryption October 2022"
subtitle: Moving to encrypted PostgreSQL storage
description: Details on upcoming maintenance period
date: 2022-10-13
authors: ["ben-hardill"]
---

# Scheduled maintenance: Database encryption October 2022

As part of an on going security [review](https://flowforge.com/product/security/#data-at-rest) of the FlowForge Managed offering we discovered that the backend database was not using encrypted storage. In keeping with industry best practices we plan to migrate the database to a new instance using encrypted at rest storage.

## Impact

> Customers' Node-RED instance will remain running, though any features that depend on FlowForge will not operate as expected. This includes user sessions, project and team management, as well as the project nodes for inter-project communication.

Reviewer: ```suggestion
Customers' Node-RED instances will remain running, though any features that depend on FlowForge will not operate as expected during the migration. This includes user sessions, project and team management, as well as the project nodes for inter-project communication.
```

> Self hosted instalations are uneffected by this change.

Reviewer: ```suggestion
Self-hosted installations are unaffected by this change.
```

## When

> The plan is to carry out this work at XX October 2022 at XX:XX UTC and the work should be complete in 2hr with the system being restored earlier if all is well.

Reviewer: ```suggestion
The migration will be on XX October 2022 at XX:XX UTC and is expected to take under 2 hours. The platform will be available as soon as the migration is complete.

We will post updates during the migration period to our Twitter account [https://twitter.com/flowforgeinc](@FlowForgeInc).
```

Reviewer 2: Love the update part on twitter, thanks for the suggestion.

---
title: "Scheduled maintenance: Database encryption October 2022"
subtitle: Moving to encrypted PostgreSQL storage
description: Details on upcoming maintenance period

> date: 2022-10-13

Reviewer: ```suggestion
date: 2022-10-18
```

authors: ["ben-hardill"]
---

# Scheduled maintenance: Database encryption October 2022

As part of an on going security [review](https://flowforge.com/product/security/#data-at-rest) of the FlowForge Managed offering we discovered that the backend database was not using encrypted storage. In keeping with industry best practices we plan to migrate the database to a new instance using encrypted at rest storage.

## Impact

Customers' Node-RED instances will remain running, though any features that depend on FlowForge will not operate as expected during the migration. This includes user sessions, project and team management, as well as the project nodes for inter-project communication.

Self-hosted installations are unaffected by this change.

## When

> The migration will be on XX October 2022 at XX:XX UTC and is expected to take under 2 hours. The platform will be available as soon as the migration is complete.

Reviewer: Don't forget to add the actual date/time :)

Author: Updated

---

> title: Planned Database Migration

Reviewer 2: Scheduled maintenance: Database encryption October 2022

Reviewer 2: l like a time stamp in the title to avoid confusion with later migrations

> subtitle: Moving to encrypted storage

Reviewer 2: ```suggestion
subtitle: Moving to encrypted PostgreSQL storage
```

> description: Details ppcoming maintence period

Reviewer 2: ```suggestion
description: Details on upcoming maintenance period
```

date: 2022-10-13
authors: ["ben-hardill"]
---

# Planned Database Migration

As part of an on going security [review](https://flowforge.com/product/security/#data-at-rest) of the FlowForge Managed offering we discovered that the backend database was not using encrypted storage. In keeping with industry best practices we plan to migrate the database to a new instance using encrypted at rest storage.

## Impact

> To do this cleanly we are going to need to schdule a short period of downtime for the core FlowForge application. During this downtime the following functions will not be available

Reviewer 2: We really need a CI task to check for typos. I'll get on that.

```suggestion
To do this cleanly we are going to need to schedule a short period of downtime for the core FlowForge application. During this downtime the following functions will not be available
```

Reviewer 2: As this list isn't exhaustive I'd flip it around actually. 

What about:

Customers' Node-RED instance will remain running, though any features that depend on FlowForge will not operate as expected. This includes user sessions, project and team management, as well as the project nodes for inter-project communication.

Reviewer 3: There's a proposal In https://github.com/flowforge/handbook/issues/129 @ZJvandeWeg

- Creating/deleting FlowForge Projects
- Creating/deleting FlowForge Users
- Deploying changes to flows running in Projects

> Existing FlowForge Projects will continue to run as normal during the downtime.

Reviewer 2: From experience: We need to stress that self managed customers will not be impacted. 

I know it's a bit redundant, but we should mention it
